src/client.cpp

```diff
@@ -637,12 +653,23 @@ int Client<IoType, SwitchDataIntegrity, SwitchActivityInfo, SwitchCycleDuration,
                     log_msg("Starting test...");
 
                     if (!g_pApp->m_const_params.pPlaybackVector) {
+                        #ifndef NO_TIMERS
                         struct itimerval timer;
                         set_client_timer(&timer);
                         if (os_set_duration_timer(timer, client_sig_handler)) {
                             log_err("Failed setting test duration timer");
                             rc = SOCKPERF_ERR_FATAL;
                         }
+                        #else
+                        set_client_timer(&timer);
```

Comment:
We should avoid calling `set_client_timer` function if we aren't going to use the timer. Seems like the function is called here in order to get the amount of time to sleep later. Why not grab the time directly in `thread_sleep` function as `set_client_timer` does?

Seconds:
`  (g_pApp->m_const_params.cooldown_msec + g_pApp->m_const_params.warmup_msec) / 1000 +
        g_pApp->m_const_params.sec_test_duration; `
Microseconds:
`(g_pApp->m_const_params.cooldown_msec + g_pApp->m_const_params.warmup_msec) % 1000;`

Reply:
Makes sense, would be nicer to cut the middle man here. Thanks, adding it.